### PR TITLE
Update tiny models for pipeline testing.

### PR DESCRIPTION
**Pull request by ydshieh**
# What does this PR do?

Update tiny models for pipeline testing.

### CODE REVIEW

File: src/transformers/__init__.py
Comment (ydshieh) [inline src/transformers/__init__.py:1]: The changes in `__init__` files and `dummy_xxx_objects.py` are fixes to make import work.

File: tests/models/git/test_modeling_git.py [hunk 1]
```diff
@@ -383,11 +383,22 @@ class GitModelTest(ModelTesterMixin, GenerationTesterMixin, PipelineTesterMixin,
     all_model_classes = (GitModel, GitForCausalLM) if is_torch_available() else ()
     all_generative_model_classes = (GitForCausalLM,) if is_torch_available() else ()
     pipeline_model_mapping = (
-        {"feature-extraction": GitModel, "text-generation": GitForCausalLM} if is_torch_available() else {}
+        {"feature-extraction": GitModel, "image-to-text": GitForCausalLM, "text-generation": GitForCausalLM}
+        if is_torch_available()
```
Comment (ydshieh) [inline tests/models/git/test_modeling_git.py:387]: Keep `"image-to-text": GitForCausalLM` here (computed in an automatic way), but skip below.
In the end, we might decide `GitForCausalLM` is not for `image-to-text` task, and we will have to remove.

File: tests/models/layoutlmv3/test_modeling_layoutlmv3.py [hunk 2]
```diff
@@ -286,10 +286,7 @@ class LayoutLMv3ModelTest(ModelTesterMixin, PipelineTesterMixin, unittest.TestCa
         else ()
     )
     pipeline_model_mapping = (
-        {
-            "document-question-answering": LayoutLMv3ForQuestionAnswering,
-            "feature-extraction": LayoutLMv3Model,
-        }
+        {"document-question-answering": LayoutLMv3ForQuestionAnswering, "feature-extraction": LayoutLMv3Model}
```
Comment (ydshieh) [inline tests/models/layoutlmv3/test_modeling_layoutlmv3.py:289]: style

File: tests/models/layoutlmv3/test_modeling_tf_layoutlmv3.py [hunk 3]
```diff
@@ -278,13 +278,7 @@ class TFLayoutLMv3ModelTest(TFModelTesterMixin, PipelineTesterMixin, unittest.Te
...
     )
     pipeline_model_mapping = (
-        {
-            "feature-extraction": TFLayoutLMv3Model,
-            "question-answering": TFLayoutLMv3ForQuestionAnswering,
-            "text-classification": TFLayoutLMv3ForSequenceClassification,
-            "token-classification": TFLayoutLMv3ForTokenClassification,
-            "zero-shot": TFLayoutLMv3ForSequenceClassification,
-        }
+        {"document-question-answering": TFLayoutLMv3ForQuestionAnswering, "feature-extraction": TFLayoutLMv3Model}
```
Comment (ydshieh) [inline tests/models/layoutlmv3/test_modeling_tf_layoutlmv3.py:281]: Those were added by users or @Rocketknight1 maybe. In any case, `pipeline_model_mapping` should be added/modified with the script `add_pipeline_model_mapping_to_test.py`.

Also, the whole pipeline test for `(TF)LayoutLMv3ModelTest` is skipped, and need a fix to re-enable it.

**Review by ydshieh — COMMENTED** [review]
Leave some comments.

Note, the changes in the attribute `pipeline_model_mapping` don't mean if the pipeline tests will be run or removed. In some cases, there are conditions to skip them. 

That attribute just means, logically, a model class is for a task and we should test it, despite we may skip them due to some reasons.

File: tests/test_pipeline_mixin.py [hunk 4]
```diff
@@ -366,6 +367,7 @@ def test_pipeline_image_segmentation(self):
     def test_pipeline_image_to_text(self):
         self.run_task_tests(task="image-to-text")
 
+    @unittest.skip(reason="`run_pipeline_test` is currently not implemented.")
```
Comment (ydshieh) [inline tests/test_pipeline_mixin.py:370]: Add this so we can see it is skipped in the log

File: tests/pipelines/test_pipelines_audio_classification.py [hunk 5]
```diff
@@ -31,9 +31,9 @@
 
 
 @is_pipeline_test
-@require_torch
```
Comment (ydshieh) [inline tests/pipelines/test_pipelines_audio_classification.py:34]: to allow to run `test_modeling_tf_wav2vec2.py -k "test_pipeline_audio_classification"`

File: tests/test_pipeline_mixin.py [hunk 6]
```diff
@@ -314,7 +315,6 @@ def data(n):
         run_batch_test(pipeline, examples)
 
     @is_pipeline_test
-    @require_torch
```
Comment (ydshieh) [inline tests/test_pipeline_mixin.py:317]: to allow to run `test_modeling_tf_wav2vec2.py -k "test_pipeline_audio_classification"`

**Review by sgugger — APPROVED** [review]
Thanks for working on this!

**Pull request merged**